tests/unit/s2n_security_policies_test.c

```diff
@@ -107,147 +107,29 @@ int main(int argc, char **argv)
             }
         }
 
-        if (has_tls_13_cipher) {
-            /* Validate that s2n_tls13_default_sig_scheme() is successful on all TLS 1.3 Security Policies for all
-             * TLS 1.3 Ciphers */
```

Comment:
I removed this test because I think it's outdated and not useful, but could use a second opinion.

The issues with this test are:
* TLS1.3 signature scheme selection doesn't consider the cipher suite, so iterating over the ciphers doesn't provide any value. Only TLS1.2 has to consider the cipher suite, since TLS1.2 cipher suites include an authentication type (RSA vs ECDSA).
* The TLS1.3 fallback logic will be successful for any security policy with at least one valid TLS1.3 signature scheme. The next test in this file checks that every policy with at least one TLS1.3 cipher suite also has at least one TLS1.3 signature scheme.

Maybe the value provided by this test was that it checks for both a valid ECDSA option and an valid RSA TLS1.3 option? But we could add that check more clearly to the test below that checks for valid TLS1.3 options.

Reply:
Sam also noted https://github.com/aws/s2n-tls/pull/4285#discussion_r1401217670. I'm going to fix that and the RSA miss in https://github.com/aws/s2n-tls/commit/669daed32040ba4208a85dbf2d1eadc30c645274 after this PR.